@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -146,20 +146,52 @@ export class PickHandler {
             this.mapView.textElementsRenderer.pickTextElements(scenePosition, pickListener);
         }
 
-        const intersects: THREE.Intersection[] = [];
         const intersectedTiles = this.getIntersectedTiles(rayCaster);
 
+        const addIntersectResults = (_intersects: THREE.Intersection[]) => {
+            for (const intersect of _intersects) {
+                pickListener.addResult(this.createResult(intersect));
+            }
+        };
+
+        const intersects: THREE.Intersection[] = [];
+
+        const intersectObjects = (tile: Tile) => {
+            intersects.length = 0;
+            rayCaster.intersectObjects(tile.objects, true, intersects);
+            addIntersectResults(intersects);
+        };
+
+        // This ensures that we check a given dependency only once (because multiple tiles could
+        // have the same dependency).
+        const checkedDependencies = new Set<number>();
+        const intersectDependentObjects = (tile: Tile) => {
+            for (const tileKey of tile.dependencies) {
+                const mortonCode = tileKey.mortonCode();
+                if (checkedDependencies.has(mortonCode)) {
+                    continue;
+                }
+                checkedDependencies.add(mortonCode);
+                const otherTile = this.mapView.visibleTileSet.getCachedTile(
+                    tile.dataSource,
+                    tileKey,
+                    tile.offset,
+                    this.mapView.frameNumber
+                );
+                if (otherTile !== undefined) {
+                    rayCaster.intersectObjects(otherTile.objects, true, intersects);
+                    addIntersectResults(intersects);
```

Comment:
This could be moved to a separate private method to improve readability.

Reply:
Agreed, will do.